libs/server/Resp/RespServerSession.cs

```diff
@@ -769,7 +769,7 @@ bool NetworkCustomProcedure()
                     return true;
                 }
 
-                TryCustomProcedure(currentCustomProcedure.CustomProcedureImpl);
+                TryCustomProcedure(customCommandManagerSession.GetCustomProcedure(currentCustomProcedure.Id, this));
```

Comment:
every call to this creates a new factory instance? that does not seem efficient at all.

Reply:
The factory invocation happens only once per session. After that it gets cached and reused similar to custom transaction.